✨Feature [#3] : 상품 단건 조회 Internal API 구현

sohyun119 opened the pull request:
## ✨ 변경 타입
* [X] 신규 기능 추가/수정
* [ ] 버그 수정
* [X] 리팩토링
* [X] 설정
* [ ] 비기능 (주석 등 기능에 영향을 주지 않음)

## ✅ 체크리스트

* [X] 코드 작성 가이드라인을 준수했는가?
* [X] 변경 사항에 대한 테스트를 완료했는가?
* [ ] 문서 변경이 필요한 경우, 해당 내용을 반영했는가?

## 🚀 변경 내용
* 상품 단건 조회 Internal API - Controller 분리 생성
* 상품 단건 조회 Internal API - Service 구현 (deleted_at is null, not null 둘 다 가능)
* BaseEntity에서 가져올때 필요한 Audit Config 추가
* 테스트 코드 작성

sohyun119 committed:
🩹 fix [#3] : 도메인 레이어에서 예외처리를 진행하여, 서비스 레이어에서 예외처리 제외하는 형식으로 변경

sohyun119 committed:
🩹 fix [#3] : 주석 삭제

sohyun119 committed:
🩹 fix [#3] : 공통모듈 업데이트 - 의존성 수정

sohyun119 committed:
🩹 fix [#3] : Internal 서버를 위해서 jpa에서 직접 deleted_at IS NULL 확인
- Sql 어노테이션에서 필터기능을 위해 다른 방법을 사용해보았습니다.
- BaseEntity를 상속받아오는 바람에 필터 어노테이션이 먹지 않습니다.

sohyun119 committed:
✨ feat [#3] : Controller 구현 - Internal 상품 단건 조회 API 분리 생성

sohyun119 committed:
✨ feat [#3] : Service 구현 - Internal 상품 단건 조회 API 분리 생성

sohyun119 committed:
✨ feat [#3] : 공통모듈에서 Auditing을 위한 Config Class 작성

sohyun119 committed:
🍻 test [#3] : Internal API 상품 단건 조회 테스트 코드 작성

chokyuseong00 commented on src/main/java/com/sparta/limited/product_service/config/ProductAuditConfig.java:
```diff
@@ -0,0 +1,11 @@
+
+@Configuration
+@Import(AuditorAwareConfig.class)
+public class ProductAuditConfig {
```
해당 부분을 공통 모듈에서 처리하는 방법도 한번 같이 찾아보면 좋을 것 같아요!

sohyun119 replied:
네!! 감사합니다~! 임시로만 생각하고 내일 또 같이 찾아볼게요 ㅎㅎㅎ

chokyuseong00 commented on src/main/java/com/sparta/limited/product_service/infrastructure/repository/ProductRepositoryImpl.java:
```diff
@@ -20,8 +20,15 @@ public void save(Product product) {
-        return jpaProductRepository.findById(productId);
+    public Product findById(UUID productId) {
+        return jpaProductRepository.findById(productId)
+            .orElseThrow(ProductNotFoundException::new);
```
서비스에서 직접 예외처리를 하는 것이 아닌 레포지토리 구현체에서 예외처리를 진행하여 서비스에서 공통적으로 사용할 수 있게 작성하신 부분이 좋은 것 같습니다!

sohyun119 replied:
히히 감사합니다!!

chokyuseong00 commented on src/test/resources/http/product-test.http:
```diff
@@ -1,13 +1,19 @@
 ### 상품 생성
 POST http://localhost:8080/api/v1/products
 Content-Type: application/json
+X-User-Id: 123
```
게이트웨이에서 X-User-Id를 보내주는것을 가정하여 테스트를 진행하셨네요! 후에 게이트웨이가 완성 되었을 경우 제대로 들어가는지 다시 확인하면 좋을 것 같아요!

sohyun119 committed:
🩹 fix [#3] : Internal API 조건 수정
 - 해당 API 에서도 Deleted At IS NOT NULL 조건을 붙이는게 나을 것 같아서 수정진행하였습니다.
 - Entity에 @SQLRestriction 어노테이션 추가
 - Service 기존 상품 단일 생성 메서드와 동일하게 변경
 - Controller productService.getProduct메서드를 불러올 수 있도록 수정
 - Repository에서 사용하지 않는 deletedAtIsNull 조건 붙인 것들 삭제

AnJungGeon reviewed (approved):
수고하셨습니다!

mummumm reviewed (approved):
고생하셨습니다 !

sohyun119 committed:
🩹 fix [#3] : common module 버전 업데이트
